R/modspec.R

```diff
@@ -195,24 +195,48 @@ check_spec_contents <- function(x, crump = TRUE, warn = TRUE, ...) {
   return(invisible(NULL))
 }
 
-audit_spec <- function(x, spec, warn = TRUE) {
-  cmt <- names(init(x))
-  if(!has_name("ODE", spec) | !warn | length(cmt) ==0) {
+grepl_dxdt_ode <- function(spec, cmt) {
+  dx <- paste0("dxdt_", cmt)
+  z <- vapply(dx, FUN.VALUE = TRUE, function(dxi) {
+    any(grepl(dxi, spec[["ODE"]], fixed = TRUE))
+  })  
+  names(z) <- dx
+  z
+}
+
+audit_spec <- function(x, spec, nmv, env, warn = TRUE) {
```

Comment:
`nmv` is an object containing all the information we have about the `nm-vars` in use (e.g. `F2`, A_0(1)`, etc. It is null if `nm-vars` isn't invoked.